### PR TITLE
Adds shard_count parameter to create_distributed_table

### PR DESCRIPTION
**Pull request by halilozanakgul**
DESCRIPTION: Adds a parameter for shard count to create_distributed_table function

### CODE REVIEW

File: src/backend/distributed/commands/create_distributed_table.c [hunk 1]
```diff
@@ -229,8 +230,15 @@ create_distributed_table(PG_FUNCTION_ARGS)
 
 	char *colocateWithTableName = text_to_cstring(colocateWithTableNameText);
 
+	if (shardCount < 1 || shardCount > 64000)
```
Comment (marcocitus) [inline src/backend/distributed/commands/create_distributed_table.c:233]: should we enforce a maximum? I think how many shards you are able to create is very machine-dependent.

File: src/backend/distributed/sql/udfs/create_distributed_table/latest.sql [hunk 2]
```diff
@@ -0,0 +1,15 @@
+DROP FUNCTION create_distributed_table(regclass, text,  citus.distribution_type, text);
+CREATE OR REPLACE FUNCTION create_distributed_table(table_name regclass,
+                                                    distribution_column text,
+                                                    distribution_type citus.distribution_type DEFAULT 'hash',
+                                                    colocate_with text DEFAULT 'default',
+                                                    shard_count int DEFAULT current_setting('citus.shard_count')::int)
```
Comment (marcocitus) [inline src/backend/distributed/sql/udfs/create_distributed_table/latest.sql:6]: Do we need to have some logic around colocate_with + shard_count?

I'm wondering whether the default should actually be NULL, which would mean:
- If auto-colocatable, disallow shard_count
- If colocate_with := '<table>', disallow shard_count
- If colocate_with := 'none' or first table in co-location group, use shard_count argument, or fall back to citus.shard_count if NULL

File: src/backend/distributed/commands/create_distributed_table.c (hunk 1 above)
Comment (halilozanakgul) [reply src/backend/distributed/commands/create_distributed_table.c:233]: These are the limits for `citus.shard_count` (and I also tried to match the error message with that)
I can remove the max.

Comment (marcocitus) [reply src/backend/distributed/commands/create_distributed_table.c:233]: ah, I did not realize these are pre-existing limits

File: src/backend/distributed/commands/create_distributed_table.c [hunk 3]
```diff
@@ -193,12 +195,37 @@ master_create_distributed_table(PG_FUNCTION_ARGS)
...
+
+	bool shardCountIsStrict = false;
+	int shardCount = ShardCount;
+	if (!PG_ARGISNULL(4))
+	{
+		if (pg_strncasecmp(colocateWithTableName, "default", NAMEDATALEN) != 0 &&
+			pg_strncasecmp(colocateWithTableName, "none", NAMEDATALEN) != 0)
+		{
+			ereport(ERROR, (errmsg("Cannot use colocate_with with a table "
+								   "and shard_count at the same time")));
```
Comment (marcocitus) [inline src/backend/distributed/commands/create_distributed_table.c:218]: would be good to add some test coverage for these cases

**Review by marcocitus — DISMISSED** [review]

**Review by marcocitus — APPROVED** [review]

**Pull request merged**